### DISCUSSION TITLE
listFilter ошибка ключ не обнаружен

### DISCUSSION THREAD

**Discussion by Re-Requeste** [totumonline/totum-ru-issues-and-discussions#554] [Как это сделать?] [answered]
Добрый день, подскажите пожалуйста куда копать, есть список который беру в телеграмм по getUpdate:(вебхук не подходит) 
Практически везде message массив, но в одном случае my_chat_member, и listFilter выдает ошибку что **"Ключ message не обнаружен в одном из элементов массива."** , не получается пропустить этот массив.

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/23a91faf-266e-48fb-a098-65ce8f04b44b)

Как пересобрать список, или же как пропускать не подходящие массивы?

Сам код: 
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/8b098100-4dd8-451a-bd92-90c6076b963a)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Это ошибка listFiltes - там есть параметр skip — true, если в key: используется item и в фильтруемых подмассивах он может отсутствовать, то пропускать элементы, в которых его не существует. В противном случае функция завершится ошибкой.

**Reply by Re-Requeste** [reply]
Добавлял skip: true , ошибка сохранилась.
Попробовал переписать: listFilter(list: $dannye[result][[message]]; key: "item" = #h_generaciya_koda; item: "text"; skip: true) , ошибка так же сохранилась

**Reply by ttm-alexeypolunin** [reply]
Там нет ключа message, а не text. Отфильтруйте сначала по наличию message

**Reply by Re-Requeste** [reply]
забыл дописать здесь chat id  (listFilter(list: $dannye[result][[message]][[chat]][[id]] ,  )
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/a7cc76a1-416f-46b6-ad3a-89b3cf102120)

Вот код: 
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/e0910b44-8fcb-4580-aac7-802edb6db8ab)
возможно я что то не так делаю? 

Если убрать фильтр для теста, и сделать так $dannye[result], он выдаст результаты, если же  $dannye[result][[message]] сразу выдает ошибку

**Reply by ttm-alexeypolunin** [reply]
Не во всех элементах массива есть коюч message, вам нужно сначала отфильтровать   По наличию этого ключа message != $#lc
